keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components/README.md

```diff
@@ -0,0 +1,239 @@
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes to deprecate and in the future to remove a subset of the klog
+command line flags from Kubernetes components to encourage more diverse 
+approaches to logging in kubernetes ecosystem logging.
+
+## Motivation
+
+Early on Kubernetes adopted glog logging library for logging. Overtime the glog
+was forked to klog and multiple improvements were implemented, but features put
+into klog only piled up and were never removed. Introduction of alternative log
+formats like JSON created a conundrum, should we implement all klog features for
+JSON? Most of them don't make sense and method for their configuration leaves
+much to be desired. Klog features are controlled by set of global flags that
+remain last bastion of global state in k/k repository. Those flags don't have a
+single naming standard (some start with log prefix, some not), don't comply to
+k8s flag naming (use underscore instead of hyphen) and many other problems. We
+need to revisit how logging configuration is done in klog so it can work with
+alternative log formats and comply with current best practices.
+
+Large number of features added to klog has lead to large drop in quality. 
```

Comment:
I don't know that we've necessarily seen a drop in quality, but we certainly have not been investing in klog to the extent needed for the surface area of supported features, and I think it's fair to call that out. I think we should call out these issues explicitly: performance/scalability, complexity, etc.

Reply:
Added more examples